DAO.sol

```diff
@@ -281,11 +281,22 @@ contract DAO is DAOInterface, Token, TokenSale {
 
     function vote(uint _proposalID, bool _supportsProposal) onlyTokenholders noEther returns (uint _voteID) {
         Proposal p = proposals[_proposalID];
-        if (p.voted[msg.sender] || now >= p.votingDeadline) throw;
+        if (p.votedYes[msg.sender] || p.votedNo[msg.sender] || now >= p.votingDeadline) throw;
+
+        if (_supportsProposal){
+            p.yea += balances[msg.sender];
+            p.votedYes[msg.sender] = true;
+        }
+        else{
+            p.nay += balances[msg.sender];
+            p.votedNo[msg.sender] = true;
+        }
+
+        if (blocked[msg.sender] == 0)
+            blocked[msg.sender] = _proposalID;
+        else if (p.votingDeadline > proposals[blocked[msg.sender]].votingDeadline)
```

Comment:
I understand the logic here, but maybe add a comment for the `else if` like below:
`If token holder is already blocked by another proposal, but this proposal lasts longer, make this the blocking proposal for the token holder`